contracts/token/oracles/OracleAggregator.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IOracleAggregator.sol";
+import "./FeedInterface.sol";
+
+abstract contract OracleAggregator is Ownable, IOracleAggregator {
+
+    error MismatchInBaseAndQuoteDecimals();
+    error InvalidPriceFromRound();
+    error LatestRoundIncomplete();
+    error PriceFeedStale();
+    error OracleAddressCannotBeZero();
+
+     struct TokenInfo {
+        uint8 tokenDecimals;
+        address tokenOracle;
+        address nativeOracle;
+        bool isDerivedFeed;
+     }
```

Comment:
errors, events and structs should be moved to interface

Reply:
ok